quint/src/repl.ts

```diff
@@ -426,7 +427,7 @@ function loadFromFile(out: writer, state: ReplState, filename: string): boolean
       const defsAndExprs = modulesAndRepl[1].split(replEnd)
 
       // save the definition history
-      newState.defsHist = defsAndExprs[0]
+      newState.defsHist += defsAndExprs[0]
```

Comment:
I am surprised by `=` changed to `+=` here. My expectation is that `.load` forget all previous definitions.

Reply:
Your expectation is absolutely correct, but this is `loadFromFile`, not `load`. `load` calls `state.clear()` before calling `loadFromFile`.

Another important thing to consider is that `state.clear()` doesn't empty `defsHist`, but set it to the initial state, which is a state containing the simulator builtins (from `newCompilationState()`), so `=` will overwrite that while `+=` will keep the builtins. 

This is something I found while debugging, not necessarily related to the actual bug. But seems like I got something wrong because we have integration tests breaking because of this. I'll look into that.

Reply:
Ah, the `__repl__` module is also written to the saved file, so those are loaded back twice now. Well, this will all change when I get rid of `__repl__` in the followup as you suggested, so I just reverted this change for now.